internal/storage/level_cache.go

```diff
@@ -0,0 +1,142 @@
+package storage
+
+import (
+	"go.uber.org/zap"
+	"golang.org/x/exp/mmap"
+
+	"github.com/milvus-io/milvus/internal/log"
+	"github.com/milvus-io/milvus/internal/util/cache"
+)
+
+type LevelCache struct {
+	memCache         *cache.LRU
+	localFileCache   *cache.LRU
+	localCacheEnable bool
+
+	memoryHitsCount     int
+	localCacheHitsCount int
+}
+
+func (l *LevelCache) Add(key cache.Key, value cache.Value) {
+	if l.localCacheEnable {
+		l.localFileCache.Add(key, value)
+		if mmapValue, ok := l.localFileCache.Get(key); ok {
+			l.memCache.Add(key, mmapValue)
+		}
+		return
+	}
+	l.memCache.Add(key, value)
+}
+
+func (l *LevelCache) Get(key cache.Key) (value cache.Value, ok bool) {
+	if v, ok := l.memCache.Get(key); ok {
+		l.memoryHitsCount++
+		return v, ok
+	}
+	if l.localCacheEnable {
+		if v, ok := l.localFileCache.Get(key); ok {
+			l.localCacheHitsCount++
+			l.memCache.Add(key, v)
+			return v, ok
+		}
+	}
+	return nil, false
+}
+
+func (l *LevelCache) Remove(key cache.Key) {
+	l.memCache.Remove(key)
```

Comment:
if mem cache in , then no need to move from localFileCache

Reply:
lgtm